app/src/main/java/com/example/triptracker/viewmodel/MapViewModel.kt

```diff
@@ -16,19 +16,25 @@ import kotlinx.coroutines.launch
  * ViewModel for the MapOverview composable. It contains the city name state and the geocoder to
  * reverse decode the location.
  */
-class MapViewModel : ViewModel() {
+class MapViewModel(
+    geocoder: NominatimApi = NominatimApi(),
+    pathList: MutableLiveData<ItineraryList> = MutableLiveData<ItineraryList>(),
+    repository: ItineraryRepository = ItineraryRepository(),
+    filteredPathList: MutableLiveData<Map<Itinerary, List<LatLng>>> =
+        MutableLiveData<Map<Itinerary, List<LatLng>>>()
+) : ViewModel() {
```

Comment:
Instead of declaring all the variables after I think it is possible to directly put "val" in front of each of them in the constructor

Reply:
even for private ones (you can see an example in the UserProfileViewModel file :)

Reply:
Yes, good catch! 
Thank you, I modify this!